internal/agent/device/image/image.go

```diff
@@ -0,0 +1,104 @@
+package image
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/container"
+)
+
+type Image struct {
+	Base   string
+	Tag    string
+	Digest string
+}
```

Comment:
Solving problems with types is the way but in this case we should really look at if we need this.